prediction_market_agent/agents/microchain_agent/nft_treasury_game/deploy_nft_treasury_game.py

```diff
@@ -84,28 +94,72 @@ def load(self) -> None:
 
         super().load()
 
+    @classmethod
+    def get_initial_system_prompt(cls) -> str:
+        system_prompt = nft_treasury_game_base_prompt(wallet_address=cls.wallet_address)
+        if cls.role == Role.buyer:
+            system_prompt += nft_treasury_game_buyer_prompt()
+        elif cls.role == Role.seller:
+            system_prompt += nft_treasury_game_seller_prompt()
+        return system_prompt
+
+    def get_holding_n_nft_keys(self) -> int:
+        contract = ContractOwnableERC721OnGnosisChain(address=NFT_TOKEN_FACTORY)
+        return contract.balanceOf(Web3.to_checksum_address(self.wallet_address))
+
     def before_iteration_callback(self) -> CallbackReturn:
         """
         In following if statements, we hard-code a few special cases about the game status, to make it a bit easier on the agent logic.
         """
         system_prompt = self.agent.history[0] if self.agent.history else None
+        is_seller_without_keys = (
+            self.role == Role.seller and not self.get_holding_n_nft_keys()
+        )
 
         # First, if the agent just started, but the game is not ready yet, make him sleep and stop -- until the game is ready.
         # Otherwise, he would try to learn from past games (until he realise there are none!), he'd try to communicate, but without any money it would fail, etc.
-        if (
-            len(self.agent.history) <= 1  # One optional system message doesn't count.
-            and get_nft_game_status() == NFTGameStatus.finished
+        if len(
+            self.agent.history
+        ) <= 1 and (  # One optional system message doesn't count.
+            is_seller_without_keys or get_nft_game_status() == NFTGameStatus.finished
         ):
             logger.info(
                 "The game is not ready yet and agent didn't have any previous interactions, sleeping and stopping."
             )
             time.sleep(60)
             return CallbackReturn.STOP
 
-        # Second, if this is the agent's first iteration after the game finished, force him to reflect on the past game.
-        elif (
-            not self.game_finished_already_detected
-            and get_nft_game_status() == NFTGameStatus.finished
+        # If agent did the reflection (from the later if-clause), then...
+        elif self.agent.history and GameRoundEnd.GAME_ROUND_END_OUTPUT in str(
```

Comment:
I added `is_seller_without_keys` to the conditions to stop the game, but otherwise it's the same thing as before, just reordered the clauses.